01-data-structures/01-introduction-to-data-structures/screen/screen.rb

```diff
@@ -6,18 +6,49 @@ class Screen
   attr_accessor :matrix
 
   def initialize(width, height)
+    @width = width
+    @height = height
+    @matrix = []
+
+    for j in (0...height)
+      row = []
+      for i in (0...width)
+        row.push(nil)
+      end
+      @matrix.push(row)
+    end
+
   end
 
   # Insert a Pixel at x, y
   def insert(pixel, x, y)
+    if self.get_pixel(x, y)
+      return "Pixel already exists in this position. Use overwrite instead"
```

Comment:
removed these text based returns. Actually, took out the overwrite function all together.